phone-book/src/contacts.js

```diff
@@ -1,39 +1,73 @@
-
-
 class Contacts {
-  constructor(containerSelector) {
-    this.container = document.body.querySelector(containerSelector);
+  constructor(app) {
+    this.title = "Contacts";
+    this.app = app;
+    this.appContainer = app.appContainer;
     this.pageName = 'Contacts';
-    this.users = users;
-    this.usersOrderDirection = 'asc';
-    this.container.addEventListener('click', (event) => {
+    this.users = [];
+    this.usersOrderDirection = 'asc';    
+  }
+
+  initEvents(){
+    this.appContainer.querySelector("main > div.container").addEventListener('click', (event) => {
       if(event.target.nodeName === 'TH'){
         let sortBy = event.target.getAttribute("data-name");
         this.users = this.sortUsersBy(sortBy);
         this.usersOrderDirection = this.usersOrderDirection == 'asc' ? 'desc' : 'asc';
-        this.insertListOfUsers();
+        this.insertListOfUsersToApp(this.users);
       }
-    });
-
-
+    }); 
+    this.getListOfUsers();  
+    this.initSearch(); 
+  }
 
+  showUserPage(id){
+    let promisWithUsers = [];
+    const fetcher = new Fetcher();
```

Comment:
Please create 1 instance of Fether and use it every where in the app.
But I don't like the idea to have an a class here.

And naming is pretty weird